From 1031aceed1d9f8bd2d0bd24a6d805ad21880e24b Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Thu, 5 Dec 2024 23:59:14 +0100
Subject: [PATCH] test_ffi: increase dummy_buffer size to 512

The code below does `std::span(dummy_buffer).first(ideal_granularity)`
which obtains an ideal_granularity-sized subslice of the dummy_buffer,
however the ideal_granularity is CPU-dependent and on some CPUs this
results in 512 and the dummy_buffer is 256. That means it tries to
obtain a slice larger than the buffer, and in our hardened setup,
crashes.
---
 src/tests/test_ffi.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/tests/test_ffi.cpp b/src/tests/test_ffi.cpp
index 647e6ad..ad65759 100644
--- a/src/tests/test_ffi.cpp
+++ b/src/tests/test_ffi.cpp
@@ -1381,7 +1381,7 @@ class FFI_AEAD_Test final : public FFI_Test {
             std::vector<uint8_t> ciphertext(ideal_granularity * pt_multiplier + taglen);
             TEST_FFI_OK(botan_rng_get, (rng, plaintext.data(), plaintext.size()));
 
-            std::vector<uint8_t> dummy_buffer(256);
+            std::vector<uint8_t> dummy_buffer(512);
             TEST_FFI_OK(botan_rng_get, (rng, dummy_buffer.data(), dummy_buffer.size()));
             std::vector<uint8_t> dummy_buffer_reference = dummy_buffer;
 
-- 
2.47.1

